lib/Models/FunctionParameter.ts

```diff
@@ -0,0 +1,41 @@
+import { observable } from "mobx";
+import isDefined from "../Core/isDefined";
+
+export interface Options {
+  id: string;
+  name?: string;
+  description?: string;
+  isRequired?: boolean;
+  converter: unknown;
+}
+
+interface Feature {
+  type: string;
+  geometry: any;
+}
+
+export default abstract class FunctionParameter {
+  abstract readonly type: string;
+  readonly id: string;
+  readonly name: string;
+  readonly description: string;
+  readonly isRequired: boolean;
+  readonly converter: unknown;
+
+  @observable value?: unknown;
+
+  geoJsonFeature?: Promise<Feature> | Feature | undefined;
```

Comment:
What is this for? I think it should be private, const, or observable

Reply:
This is for sub classes to return a geojson feature representing them. Ideally it should be an optional `abstract`, but there is no such thing in TS. I'll make it `readonly`.